roles/postgres/molecule/default/verify.yml

```diff
@@ -0,0 +1,74 @@
+
+---
+- name: postgres Tests
+  hosts: all
+  become: yes
+  vars:
+    postgres_major_version: "15"
+  #   activemq_version: "5.18.7"
+  #   activemq_binary_path: "/opt/apache-activemq-{{ activemq_version }}/bin/activemq"
+  #   config_folder: "/etc/opt/alfresco"
+  #   activemq_password: "activemq"
+  tasks:
+  - name: Gather service facts
+    service_facts:
+  # - name: Print ActiveMQ password
+  #   debug:
+  #     msg: "ActiveMQ password is: {{ inventory_hostname }}"
+  - name: Set PostgreSQL service name based on OS type
+    set_fact:
+      service_name: >-
+        {{ 'postgresql' if not ansible_facts['distribution_file_variety'] == 'RedHat'
+          else 'postgresql-' ~ postgres_major_version }}
+
+  - name: Ensure PostgreSQL service is running and enabled
+    fail:
+      msg: "PostgreSQL service is not running or enabled!"
+    when: ansible_facts.services[service_name ~ '.service'].state != "running" or ansible_facts.services[service_name ~ '.service'].status != "enabled"
+  - name: Check if alfresco and alfresco-sync databases exist
+    shell: |
+      psql -U postgres -d alfresco -c 'SELECT 1'
+      psql -U postgres -d alfresco-sync -c 'SELECT 1'
+    become: yes
+    become_user: postgres
```

Comment:
You can use a [block](https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_blocks.html#grouping-tasks-with-blocks) statement to "propagate" `become_user" to all tasks which need it and avoid repeating it